test/compile-test.cc

```diff
@@ -171,3 +174,71 @@ TEST(CompileTest, TextAndArg) {
   EXPECT_EQ("42!", fmt::format(FMT_COMPILE("{}!"), 42));
 }
 #endif
+
+#if __cplusplus >= 202002L
+template <size_t max_string_length> struct test_string {
+  template <typename T> constexpr bool operator==(const T& rhs) const noexcept {
+    return (std::string_view(rhs).compare(buffer.data()) == 0);
+  }
+
+  std::array<char, max_string_length> buffer{};
+};
+
+template <size_t max_string_length, typename... Args>
+consteval auto test_format(auto format, const Args&... args) {
+  test_string<max_string_length> string{};
+  fmt::format_to(string.buffer.data(), format, args...);
+  return string;
+}
+
+TEST(CompileTimeFormattingTest, Bool) {
+  {
+    constexpr auto result = test_format<5>(FMT_COMPILE("{}"), true);
+    EXPECT_EQ(result, "true");
```

Comment:
Now that `test_format` is consteval we don't need to store the result in a constexpr variable. This can be just 

```c++
    EXPECT_EQ(test_format<5>(FMT_COMPILE("{}"), true), "true");
```

And the blocks can be removed.

Reply:
Actually, it doesn't work because of `EXPECT_EQ` internals, for example, this code on Clang produces:
```
test/compile-test.cc:196:30: error: lambda expression in an unevaluated operand
    EXPECT_EQ(test_format<5>(FMT_COMPILE("{}"), true), "true");
                             ^
include/fmt/compile.h:38:24: note: expanded from macro 'FMT_COMPILE'
#define FMT_COMPILE(s) FMT_STRING_IMPL(s, fmt::detail::compiled_string)
                       ^
include/fmt/format.h:3246:3: note: expanded from macro 'FMT_STRING_IMPL'
  [] {                                                                     \
```

Reply:
But code with `_cf` literal probably will 😏

Reply:
OK, let's keep as is then.